src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -82,4 +82,19 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+
+    /**
+     * Concatenate printables passed in as variables
+     * @return String with concatenated printables
+     */
+    default String getPrintableString(Printable... printables){
+
+        StringBuilder sb = new StringBuilder();
+
+        for(Printable p : printables) {
+            sb.append(p.getPrintableString());
```

Comment:
Wouldn't this result in an ugly string with no separator between each part? 😝

Reply:
:bulb: *Java tip*: Java 8 has a `StringJoiner` class that can join multiple strings using a delimiter e.g. `aaa` `bbb` `ccc` --> `aaa, bbb, ccc` https://www.mkyong.com/java8/java-8-stringjoiner-example/

Reply:
I have put the spaces in the defined prefixes. 
E.g. public static final String ADDRESS_PRINTABLE_PREFIX = " Address: ";

Would it be better to have it in code?